infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -313,6 +356,63 @@ private boolean runRecoveryReconfiguration() {
         return recoveryResult;
     }
 
+    /**
+     * Task to collect local node metrics.
+     * This task collects the following:
+     * Boolean Status of layout, sequencer and logunit servers.
+     * Logunit Metrics:
+     * Size of writeQueue for pending write requests.
+     * These metrics are then composed into NodeMetrics protobuf model
+     * and stored locally.
+     */
+    private void updateLocalMetrics() {
+        // Initializing the status of components
+        // No need to poll unless node is bootstrapped.
+        Layout layout = serverContext.getManagementLayout();
+        if (layout == null) {
+            return;
+        }
+
+        // Build and replace existing locally stored nodeMetrics
+        SequencerMetrics sequencerMetrics;
+        LogUnitMetrics logUnitMetrics;
+        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
+            try {
+                sequencerMetrics = CFUtils.getUninterruptibly(
+                        getCorfuRuntime()
+                                .getRouter(getLocalEndpoint())
+                                .getClient(SequencerClient.class)
+                                .requestMetrics());
+            } catch (ServerNotReadyException snre) {
+                sequencerMetrics = SequencerMetrics.newBuilder()
+                        .setSequencerReady(false)
+                        .build();
+            } catch (Exception e) {
+                log.error("Error while requesting metrics from the sequencer: ", e);
+                sequencerMetrics = SequencerMetrics.getDefaultInstance();
+            }
+        } else {
+            sequencerMetrics = SequencerMetrics.getDefaultInstance();
+        }
+
+        try {
+            logUnitMetrics = CFUtils.getUninterruptibly(
```

Comment:
For chain replication it might be a good idea to have a metric for disk write latency. The queue size might not be able to reflect much in chain replication.